apps/laboratory/src/components/Solana/SolanaWriteContractTest.tsx

```diff
@@ -26,7 +26,7 @@ export function SolanaWriteContractTest() {
     const PROGRAM_ID = new PublicKey(detectProgramId(currentChain?.chainId ?? ''))
 
     try {
-      if (!walletProvider || !address) {
+      if (!walletProvider?.publicKey) {
```

Comment:
@zoruka Is this the only way to check if user is connected / have address? Would this be something unified for other chains as well? (AFAIK the `publicKey` is a concept only on Solana)

Reply:
This is related to type assertion for `publicKey` , not exactly to check if is connected or not. To check the connection status we should use the `isConnected` method from `useWeb3ModalAccount`.

Reply:
Btw, I have removed the address checking because it was not in use besides this conditional.